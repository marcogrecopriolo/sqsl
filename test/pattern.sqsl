{ simple delimited - populate }
display "output 10 to 50";
output to "patternin.res";
for  i in 10, 20, 30, 40, 50
do
        display i, i+1, i+2 format delimited;
end for;
append to default;

display "input pattern delimited";
input from "patternin.res" pattern delimited;

display "input pattern delimited by";
input from "patternin.res" pattern delimited by "|";

display "input pattern delimited by | escape \\";
input from "patternin.res" pattern delimited by "|" escape "\\";

display "output 10 to 50";
output to "patternin.res";
for  i in 10, 20, 30, 40, 50
do
        display i, i+1, ascii(i+64), i+2 format full "%i;%i;%s>;%i";
end for;
append to default;

display "input pattern delimited by ; escape >";
input from "patternin.res" pattern delimited by ";" escape ">";

{ split }
display "split delimited";
display split "10|20|30|" pattern delimited;

display "split delimited by";
display split "10|20|30|" pattern delimited by "|";

display "split delimited by | escape \\";
display split "10|20|30|" pattern delimited by "|" escape "\\";

display "split delimited by ; escape >";
display split "10;20>;30;" pattern delimited by ";" escape ">";
