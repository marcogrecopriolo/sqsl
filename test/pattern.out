output 10 to 50
input pattern delimited
10,11,12
20,21,22
30,31,32
40,41,42
50,51,52
input pattern delimited by
10,11,12
20,21,22
30,31,32
40,41,42
50,51,52
input pattern delimited by | escape \
10,11,12
20,21,22
30,31,32
40,41,42
50,51,52
output 10 to 50
input pattern delimited by ; escape >
10,11,J;12
20,21,T;22
30,31,^;32
40,41,h;42
50,51,r;52
split delimited
10,20,30,
split delimited by
10,20,30,
split delimited by | escape \
10,20,30,
split delimited by ; escape >
10,20;30,
